src/UserInterface/templates/Number_Guesser.html

```diff
@@ -0,0 +1,118 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Number Guesser</title>
+    <style>
+        body {
+            font-family: Arial, sans-serif;
+            text-align: center;
+            padding: 20px;
+        }
+        .hidden {
+            display: none;
+        }
+        #message {
+            margin: 20px 0;
+            font-size: 1.2em;
+            color: #333;
+        }
+        input, button {
+            padding: 10px;
+            margin: 10px;
+            font-size: 1em;
+        }
+    </style>
+</head>
+<body>
+    <h1>Number Guesser</h1>
+    <div id="join">
+        <input type="text" id="player_id" placeholder="Enter Player ID">
+        <button onclick="joinGame()">Join Game</button>
+    </div>
+
+    <div id="set-number" class="hidden">
+        <input type="number" id="target_number" min="1" max="10" placeholder="Number (1-10)">
+        <button onclick="setNumber()">Set Number</button>
+    </div>
+
+    <div id="guess-number" class="hidden">
+        <input type="number" id="guess" min="1" max="10" placeholder="Guess the Number">
+        <button onclick="guessNumber()">Guess</button>
+    </div>
+
+    <div id="message"></div>
+
+    <script src="https://cdn.socket.io/4.6.1/socket.io.min.js"></script>
+    <script>
+        const socket = io();
```

Comment:
Should be:
var socket = io.connect('http://' + document.domain + ':' + location.port);